src/map.c

```diff
@@ -131,6 +132,11 @@ void map_cleanup(void)
  */
 gboolean map_keypress(GtkWidget *widget, GdkEventKey* event, gpointer data)
 {
+    if (is_processing_events()) {
+        // events are processing, pass all keys unmodified
+        return false;
```

Comment:
I think we have to queue this event too, else we could end up with key events put into wrong order - current one followed by some of the enqueued events.